cleverhans/experimental/certification/optimization.py

```diff
@@ -88,75 +176,76 @@ def get_min_eig_vec_proxy(self, use_tf_eig=False):
     # Using autograph to automatically handle
     # the control flow of minimum_eigen_vector
     min_eigen_tf = autograph.to_graph(utils.minimum_eigen_vector)
+
     def _vector_prod_fn(x):
       return self.dual_object.get_psd_product(x)
+
     estimated_eigen_vector = min_eigen_tf(
         x=self.eig_init_vec_placeholder,
         num_steps=self.eig_num_iter_placeholder,
         learning_rate=self.params['eig_learning_rate'],
         vector_prod_fn=_vector_prod_fn)
     return estimated_eigen_vector
 
-  def prepare_one_step(self):
+  def prepare_for_optimization(self):
     """Create tensorflow op for running one step of descent."""
-    # Create the objective
 
-    self.dual_object.set_differentiable_objective()
     self.eig_vec_estimate = self.get_min_eig_vec_proxy()
     self.stopped_eig_vec_estimate = tf.stop_gradient(self.eig_vec_estimate)
     # Eig value is v^\top M v, where v is eigen vector
-    self.eig_val_estimate = tf.matmul(tf.transpose(
-        self.stopped_eig_vec_estimate),
-                                      self.dual_object.get_psd_product(
-                                          self.stopped_eig_vec_estimate))
+    self.eig_val_estimate = tf.matmul(
+        tf.transpose(self.stopped_eig_vec_estimate),
+        self.dual_object.get_psd_product(self.stopped_eig_vec_estimate))
     # Penalizing negative of min eigen value because we want min eig value
     # to be positive
-    self.total_objective = (self.dual_object.unconstrained_objective +
-                            0.5*(tf.square(
-                                tf.maximum(-1*self.penalty_placeholder*
-                                           self.eig_val_estimate, 0))))
+    self.total_objective = (
+        self.dual_object.unconstrained_objective + 0.5 * (tf.square(
```

Comment:
I think it would be better for readability to move `0.5*tf.square` onto next line.
Also parenthesis around tf.square are not necessary.
And usually you can replace `-1 * x` with `-x`, this will help to save few additional characters


```python
   self.total_objective = (
        self.dual_object.unconstrained_objective
        + 0.5 * tf.square(
            tf.maximum(-self.penalty_placeholder * self.eig_val_estimate, 0)))
```